### DISCUSSION TITLE
User check usage error | مشکل بررسی کاربر - حجم مصرفی و مانده

### DISCUSSION THREAD

**Discussion by geeekboy92** [hiddify/Hiddify-Manager#4210] [Bug باگ: please use issue]
سلام من پنل رو نصب کردم ولی هر کاربری که میسازم هرچقدر مصرف کنه از حجمش کم نمیشه حتی دکمه update زیر باکس لیست کاربرا رو میزنم ارور زیر میاد. کسی راه حلی داره ؟‌مجدد هم نصب کردم ولی نشد.

        
Command '['sudo', '/opt/hiddify-manager/common/commander.py', 'update-wg-usage']' returned non-zero exit status 1.
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/flask/app.py", line 870, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.10/dist-packages/flask/app.py", line 855, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "/usr/local/lib/python3.10/dist-packages/flask_classful.py", line 303, in proxy
    response = view(**request.view_args)
  File "/usr/local/lib/python3.10/dist-packages/flask_classful.py", line 271, in inner
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/auth.py", line 114, in decorated_view
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/panel/admin/Actions.py", line 190, in update_usage
    out_msg=f'<pre class="ltr">{json.dumps(usage.update_local_usage(),indent=2)}</pre>',
  File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/panel/usage.py", line 16, in update_local_usage
    res = user_driver.get_users_usage(reset=True)
  File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/drivers/user_driver.py", line 21, in get_users_usage
    all_usage = driver.get_all_usage(users)
  File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/drivers/wireguard_api.py", line 98, in get_all_usage
    all_usages = self.__sync_local_usages()
  File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/drivers/wireguard_api.py", line 44, in __sync_local_usages
    wg_usage = self.__get_wg_usages()
  File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/drivers/wireguard_api.py", line 22, in __get_wg_usages
    raw_output = commander(Command.update_wg_usage, run_in_background=False)
  File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/panel/run_commander.py", line 85, in commander
    return subprocess.check_output(base_cmd, cwd=str(current_app.config['HIDDIFY_CONFIG_PATH'])).decode()
  File "/usr/lib/python3.10/subprocess.py", line 421, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['sudo', '/opt/hiddify-manager/common/commander.py', 'update-wg-usage']' returned non-zero exit status 1.

**Comment by TomCat-Hub** [comment]
> سلام من پنل رو نصب کردم ولی هر کاربری که میسازم هرچقدر مصرف کنه از حجمش کم نمیشه حتی دکمه update زیر باکس لیست کاربرا رو میزنم ارور زیر میاد. کسی راه حلی داره ؟‌مجدد هم نصب کردم ولی نشد.
> 
> Command '['sudo', '/opt/hiddify-manager/common/commander.py', 'update-wg-usage']' returned non-zero exit status 1. Traceback (most recent call last): File "/usr/local/lib/python3.10/dist-packages/flask/app.py", line 870, in full_dispatch_request rv = self.dispatch_request() File "/usr/local/lib/python3.10/dist-packages/flask/app.py", line 855, in dispatch_request return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args) # type: ignore[no-any-return] File "/usr/local/lib/python3.10/dist-packages/flask_classful.py", line 303, in proxy response = view(**request.view_args) File "/usr/local/lib/python3.10/dist-packages/flask_classful.py", line 271, in inner return fn(*args, **kwargs) File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/auth.py", line 114, in decorated_view return fn(*args, **kwargs) File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/panel/admin/Actions.py", line 190, in update_usage out_msg=f'
> 
> {json.dumps(usage.update_local_usage(),indent=2)}
> ',
> File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/panel/usage.py", line 16, in update_local_usage
> res = user_driver.get_users_usage(reset=True)
> File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/drivers/user_driver.py", line 21, in get_users_usage
> all_usage = driver.get_all_usage(users)
> File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/drivers/wireguard_api.py", line 98, in get_all_usage
> all_usages = self.__sync_local_usages()
> File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/drivers/wireguard_api.py", line 44, in __sync_local_usages
> wg_usage = self.__get_wg_usages()
> File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/drivers/wireguard_api.py", line 22, in __get_wg_usages
> raw_output = commander(Command.update_wg_usage, run_in_background=False)
> File "/usr/local/lib/python3.10/dist-packages/hiddifypanel/panel/run_commander.py", line 85, in commander
> return subprocess.check_output(base_cmd, cwd=str(current_app.config['HIDDIFY_CONFIG_PATH'])).decode()
> File "/usr/lib/python3.10/subprocess.py", line 421, in check_output
> return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
> File "/usr/lib/python3.10/subprocess.py", line 526, in run
> raise CalledProcessError(retcode, process.args,
> subprocess.CalledProcessError: Command '['sudo', '/opt/hiddify-manager/common/commander.py', 'update-wg-usage']' returned non-zero exit status 1.

درود بر تو دوست عزیز
وقت بخیر

خیلی ممنونم که کامنت گذاشتید و مشکل خودتون رو با ما درمیون گذاشتید. در حال حاضر تیم توسعه از مشکل اکانتینگ در نسخه ی آخر ریلز آگاهی کامل دارند و مشکل در حال برسی و رفع میباشد. تا رفع کامل مشکل، صمیمانه از شما درخواست میکنیم که یکبار از هیدیفای منیجر خود بکاپ بگیرید و سرور خودتون رو ریبیلد کنید.

لطفا اگر سوال بیشتری داشتید و یا نیاز به راهنمایی دقیق تری بود در گروه تلگرام ما پیام بفرستید.

لینک گروه تلگرام هیدیفای:
https://t.me/hiddify_board

باتشکر از صبر و انتخاب شما
مانا باشید
